src/DDTrace/Integrations/PHPRedis/PHPRedisSandboxedIntegration.php

```diff
@@ -0,0 +1,316 @@
+        self::traceMethodAsCommand('_serialize');
+
+        self::traceMethodAsCommand('isConnected');
+        self::traceMethodAsCommand('getHost');
+        self::traceMethodAsCommand('getPort');
+        self::traceMethodAsCommand('getDbNum');
+        self::traceMethodAsCommand('getTimeout');
+        self::traceMethodAsCommand('getReadTimeout');
+
+        return SandboxedIntegration::LOADED;
+    }
+
+    public static function enrichSpan(SpanData $span, $method = null)
+    {
+        $span->service = 'phpredis';
+        $span->type = Type::REDIS;
+        if (null !== $method) {
+            // method names for internal functions are lowered so we need to explitly set them if we want to have the
+            // proper case.
+            $span->name = $span->resource = "Redis.$method";
+        }
+    }
+
+    public static function traceMethodNoArgs($method)
+    {
+        \DDTrace\trace_method('Redis', $method, function (SpanData $span, $args) use ($method) {
+            PHPRedisSandboxedIntegration::enrichSpan($span, $method);
+        });
+    }
+
+    public static function traceMethodAsCommand($method)
+    {
+        \DDTrace\trace_method('Redis', $method, function (SpanData $span, $args) use ($method) {
+            PHPRedisSandboxedIntegration::enrichSpan($span, $method);
+            $normalizedArgs = PHPRedisSandboxedIntegration::normalizeArgs($args);
+            // Obfuscable methods: see https://github.com/DataDog/datadog-agent/blob/master/pkg/trace/obfuscate/redis.go
+            $span->meta[Tag::REDIS_RAW_COMMAND]
+                = empty($normalizedArgs) ? $method : ($method . ' ' . $normalizedArgs);
+        });
+    }
+
+    /**
+     * Based on logic from python tracer:
+     * https://github.com/DataDog/dd-trace-py/blob/0d7e7cb38216acb0c8b29f0ae1318d25bc160123/ddtrace/contrib/redis/util.py#L25
+     *
+     * @param array $args
+     * @return string
+     */
+    public static function normalizeArgs($args)
+    {
+        $rawCommandParts = [];
+
+        $totalArgsLength = 0;
+        foreach ($args as $arg) {
+            if ($totalArgsLength > self::CMD_MAX_LEN) {
+                break;
+            }
+
+            $partValue = null;
+
+            if (\is_string($arg)) {
+                $partValue = $arg;
+            } elseif (\is_numeric($arg)) {
+                $partValue = (string)$arg;
+            } elseif (\is_null($arg)) {
+                $partValue = 'null';
+            } elseif (\is_bool($arg)) {
+                $partValue = $args ? 'true' : false;
+            } elseif (\is_array($arg)) {
+                // This is best effort as specific index might be missing or be shifted, e.g. [0 => 'a', 2 => 'b'].
+                // In this case the worst that can happen is that we generate '0 a 2 b' instead of 'a b'. We accept this
+                // in order to keep things as simple as possible.
+                $isAssociative = array_values($arg) !== $arg;
```

Comment:
Very good call. Worst that can happen is that we miss to add some keys, so it is worth the change. Thanks Sammy.

Reply:
I am having double thoughts about this, as many times the key is numeric and would be easily confused.

I proposed a slightly more complex solution that might serve us decently in this case.

```function isAssociative($array)
{
    $expected = 0;
    foreach ($array as $key => $value) {
        if ($key !== $expected++) {
            return true;
        }
    }
    return false;
}

var_dump(isAssociative([ 'a', 'b' ])); // false
var_dump(isAssociative([ 'a' => 1, 'b' => 2 ])); // true
var_dump(isAssociative([ '0' => 1, '2' => 2 ])); // true: false positive
var_dump(isAssociative([ '0' => 1, '1' => 2 ])); // false
```

Reply:
Brilliant! Are we expecting string keys that are sequential ints to be considered numeric arrays? `[ '0' => 1, '1' => 2 ]` or only true int keys? `[ 0 => 1, 1 => 2 ]` Just asking because in the snippet the `foreach` has a strict comparison on `$key !== $expected++`.

Nice solution! 👍

Reply:
> Are we expecting string keys that are sequential ints to be considered numeric arrays?

In my mind they should be associative...because if they are string then I expect them to be put there on purpose. But I don't have strong opinions about it.